components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/util/MicroIntegratorBaseUtils.java

```diff
@@ -607,4 +618,44 @@ public static SynapseEnvironment getSynapseEnvironment() {
         return (SynapseEnvironment) synapseEnvironmentParatemer.getValue();
     }
 
+    /**
+     * Gets Server hostname.
+     *
+     * @return SynapseEnvironment - SynapseEnvironment
+     */
+    public static String getServerHostName() {
+        return serverConfigurationService.getFirstProperty("HostName");
+    }
+
+    /**
+     * Gets Server http listener port.
+     *
+     * @return http port
+     * @throws ResolverException exception
+     */
+    public static int getServerHTTPListenerPort() throws ResolverException {
+        try {
+            return Integer.parseInt(carbonAxisConfigurator.getAxisConfiguration().getTransportsIn()
+                    .get(org.wso2.micro.core.Constants.HTTP_TRANSPORT)
+                    .getParameter(org.wso2.micro.core.Constants.TRANSPORT_PORT).getValue().toString());
```

Comment:
Shall we resolve the port offset and return?

Reply:
Fixed